internal/pkg/gateway/commit/finder.go

```diff
@@ -12,13 +12,20 @@ import (
 	"context"
 
 	"github.com/hyperledger/fabric-protos-go/peer"
+
 	"github.com/pkg/errors"
 )
 
+type Status struct {
+	BlockNumber   uint64
+	TransactionID string
+	Code          peer.TxValidationCode
+}
+
 // QueryProvider provides status of previously committed transactions on a given channel. An error is returned if the
 // transaction is not present in the ledger.
 type QueryProvider interface {
-	TransactionStatus(channelName string, transactionID string) (peer.TxValidationCode, error)
+	TransactionStatus(channelName string, transactionID string) (peer.TxValidationCode, uint64, error)
 }
```

Comment:
This describes an interface for a subset of capability provided by the PeerAdapter struct, which allows me to use a mock implementation in unit tests. Trying to make the PeerAdapter work purely with interfaces (that I could then mock) would be even more convoluted because of the way to Go type system works

Reply:
Decoupling code within a package for the sake of using mocks for UTs does not sound a good idea to me. I am not sure I understand what prevents from writing UTs as long as external dependencies are mocked.

Reply:
QueryProvider decouples the PeerAdapter (i.e. the thin layer of code not easily untangled from the peer implementation, and so you could think of as being the external dependency) from the implementation in this package